handler/category-dimension-import.go

```diff
@@ -183,3 +154,45 @@ func (h *CategoryDimensionImport) instanceFailed(ctx context.Context, err error,
 	}
 	return err
 }
+
+// onLastDimension handles the case where all dimensions have been updated to the instance.
+// If the eTag did not change since the last update, then we know that this consumer was the last one to update the instance. In that case:
+// - Set instance to edition-confirmed
+// - Set import job to completed state
+// - send an InstanceComplete kafka message
+func (h *CategoryDimensionImport) onLastDimension(ctx context.Context, e *event.CategoryDimensionImport, eTag string) error {
+
+	// set instance to 'edition-confirmed' state, only if the eTag value did not change
+	_, err := h.datasets.PutInstanceState(ctx, h.cfg.ServiceAuthToken, e.InstanceID, dataset.StateEditionConfirmed, eTag)
+	if err != nil {
+		switch putErr := err.(type) {
+		case *dataset.ErrInvalidDatasetAPIResponse:
+			if putErr.Code() == http.StatusConflict {
+				log.Info(ctx, "instance state could not be set to 'edition-confirmed' because it changed between putting the last dimension option and counting")
+				return nil // valid scenario, another consumer was the last one to update the post instances.
+			}
+		}
+		return &Error{
+			err:     fmt.Errorf("error while trying to set the instance to edition-confirmed state: %w", err),
+			logData: log.Data{"event": e},
+		}
+	}
+
+	// Import api update job to completed state
+	if err := h.importApi.UpdateImportJobState(ctx, e.JobID, h.cfg.ServiceAuthToken, StateImportCompleted); err != nil {
+		return fmt.Errorf("error updating import job to completed state: %w", err)
```

Comment:
So what happens to clean up this failed situation ?

Reply:
I think this should be retried. As it is now, it will just leave the import job in a submitted state, but the work was actually completed.

Reply:
I understand that this is the work that Nathan is investigating?

Reply:
Shouldn't the action that is being attempted on line 182 be retried a number of times with random exponential back off ?
Similar to what you have recently added elsewhere.

Reply:
In this case, I think the retries would be part of the dp-http client retry logic, I will doulbe check.

Reply:
I checked dp-net:
https://github.com/ONSdigital/dp-net/blob/ca97fb3aaaf9a715518a7d3eaf74d37218fd152e/http/client.go#L168
The client retries a call if the status code is 5xx and on Conflict (which, in this case, it does not make sense to retry with the same If-Match)